src/main/java/cuchaz/enigma/bytecode/translators/TranslationMethodVisitor.java

```diff
@@ -1,28 +1,33 @@
 package cuchaz.enigma.bytecode.translators;
 
-import cuchaz.enigma.translation.LocalNameGenerator;
 import cuchaz.enigma.translation.Translator;
 import cuchaz.enigma.translation.representation.MethodDescriptor;
 import cuchaz.enigma.translation.representation.Signature;
 import cuchaz.enigma.translation.representation.TypeDescriptor;
 import cuchaz.enigma.translation.representation.entry.*;
 import org.objectweb.asm.*;
 
-import java.util.List;
-import java.util.stream.Collectors;
-
 public class TranslationMethodVisitor extends MethodVisitor {
-	private final ClassDefEntry ownerEntry;
 	private final MethodDefEntry methodEntry;
 	private final Translator translator;
 
-	private boolean hasParameterMeta;
+	private int parameterIndex = 0;
+	private int parameterLvIndex;
 
 	public TranslationMethodVisitor(Translator translator, ClassDefEntry ownerEntry, MethodDefEntry methodEntry, int api, MethodVisitor mv) {
 		super(api, mv);
 		this.translator = translator;
-		this.ownerEntry = ownerEntry;
 		this.methodEntry = methodEntry;
+
+		parameterLvIndex = methodEntry.getAccess().isStatic() ? 0 : 1;
```

Comment:
Should this account for enum offset too like  `getArgumentIndex` did?

Reply:
No, the synthetic enum constructor parameters are visited by `visitParameter`, and the initial `parameterLvIndex` needs to be the offset of the first parameter.

Reply:
Does this affect naming though?

Reply:
It only affects the indices of the automatically generated parameter names. The remapped parameter names are still obtained, as before, by creating a `LocalVariableEntry` with the LV index of the parameter.